File: Problems/Algorithms/647. Palindromic Substrings/PalindromicSubstrings.kt
class Solution {
    fun countSubstrings(s: String): Int {
        val n = s.length
        if (n == 1) return 1
        
        var ans = 0
        for (i in 0..n-2) {
            ans++
            
            var left = i - 1
            var right = i + 1
            while (left >= 0 && right <= n-1 && s.get(left) == s.get(right)) {
                ans++
                left--
                right++
            }
            
            left = i
            right = i + 1
            while (left >= 0 && right <= n-1 && s.get(left) == s.get(right)) {
                ans++
                left--
                right++
            }
        }
        ans++
        
        return ans
    }
}

File: Problems/Algorithms/647. Palindromic Substrings/palindromic_substrings.py
#!/usr/bin/env python3


class Solution:
    def countSubstrings(self, s: str) -> int:
        n = len(s)
        
        def count(left, right):
            ans = 0
            while left >= 0 and right < n and s[left] == s[right]:
                left -= 1
                right += 1
                ans += 1
            return ans

        ans = 0
        for i in range(n):
            even = count(i, i+1)
            odd = count(i, i)
            ans += even + odd
        
        return ans
